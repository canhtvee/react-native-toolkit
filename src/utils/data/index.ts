export * from './Fetching';
export * from './Session';
