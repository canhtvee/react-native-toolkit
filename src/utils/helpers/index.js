export * from './Convert';
export * from './ResetFunction';
export * from './Validate';
