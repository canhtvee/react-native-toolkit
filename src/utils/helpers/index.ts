export * from './Convert';
export * from './ResetFunction';
export * from './Mocking';
export * from './Constants';
