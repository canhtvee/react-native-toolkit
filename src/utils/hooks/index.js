export * from './useFocusRefetch';
export * from './useRerender';
