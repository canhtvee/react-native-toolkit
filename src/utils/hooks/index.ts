export * from './useFocusRefetch';
export * from './useRerender';
export * from './useUnmountEffect';
export * from './useSetTimeout';
