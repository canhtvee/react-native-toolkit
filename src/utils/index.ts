export * from './data';
export * from './helpers';
export * from './hooks';
export * from './modules';
export * from './resources';
export * from './types';
