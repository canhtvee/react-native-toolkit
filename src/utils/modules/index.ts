export * from './Session';
export * from './AppContext';
export * from './Fetching';
export * from './Navigation';
