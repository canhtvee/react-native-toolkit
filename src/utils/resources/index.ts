export * from './images';
export * from './apis';
export * from './colors';
export * from './sizes';
export * from './styles';
export * from './strings';
export * from './system';
