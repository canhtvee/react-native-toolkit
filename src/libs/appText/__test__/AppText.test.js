import React from 'react';
import {render, cleanup, fireEvent} from '@testing-library/react-native';

import {AppContextProvider} from '../../../utils';

import {AppText} from '../index';

/**
 * Run test command
 * yarn test -u -t="AppText"
 */

afterAll(cleanup);

describe('AppText', () => {
  test('should render correctly', () => {
    const tree = render(
      <AppContextProvider>
        <AppText style={{color: 'black'}}>should render correctly</AppText>{' '}
      </AppContextProvider>,
    );

    expect(tree).toMatchSnapshot();
  });

  test('should invoke onPressMock if pressed', () => {
    const onPressMock = jest.fn();

    const {getByText} = render(
      <AppContextProvider>
        <AppText style={{color: 'black'}} onPress={onPressMock}>
          onPress
        </AppText>
      </AppContextProvider>,
    );

    const textElement = getByText('onPress');
    fireEvent.press(textElement);
    expect(onPressMock).toHaveBeenCalled();
  });
});
