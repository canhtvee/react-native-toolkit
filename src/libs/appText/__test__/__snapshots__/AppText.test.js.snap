// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppText should render correctly 1`] = `
Array [
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontSize": 28,
        },
        Object {
          "color": "black",
        },
      ]
    }
  >
    should render correctly
  </Text>,
  " ",
]
`;
